### PR TITLE
[TASK] api v.1

### PR DESCRIPTION
**Pull request by anhnt-1044**
- Tìm hiểu api
- Using: 
[grape-entity](https://github.com/ruby-grape/grape-entity): Hỗ trợ để viết api 
[grape](https://github.com/ruby-grape/grape): Xem các routes api
[grape_on_rails_routes](https://github.com/syedmusamah/grape_on_rails_routes): Hỗ trợ việc xử lý các attributes ở response trả về

### CODE REVIEW

**Comment by anhnt-1044** [comment]
ready

**Comment by anhnt-1044** [comment]
ready

**Comment by anhnt-1044** [comment]
ready

File: app/controllers/api/v1/users.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+      resource :users do
+        desc "Return all users"
+        get "", root: :users do
+          users = User.includes :microposts
+          formatted_users = Entities::User.represent users
+          response = {
+            status: "success",
+            users: formatted_users.as_json
+          }
```
Comment (manhnt-0960) [inline app/controllers/api/v1/users.rb:22]: build thêm 1 helper method respond success

**Comment by anhnt-1044** [comment]
ready

**Comment by anhnt-1044** [comment]
ready

File: app/controllers/api/v1/users.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        desc "Return a user"
+        params do
+          requires :id, type: String, desc: "ID of the user"
+        end
+        get ":id", root: "user" do
+          user = User.find params[:id]
+          format_user = Entities::User.represent user,
+                                                 only: %i(name email microposts)
+          if user.present?
```
Comment (manhnt-0960) [inline app/controllers/api/v1/users.rb:31]: cần gì check present, bắt exception not found record rồi mà

**Comment by anhnt-1044** [comment]
ready

**Review by manhnt-0960 — COMMENTED** [review]
merged

**Pull request merged**